File: Python/gera_cpf.py
import random

def calcula_digito(digitos):
    soma = 0
    for i, digito in enumerate(digitos):
        soma += int(digito) * (len(digitos) + 1 - i)
    resto = soma % 11
    return str(0 if resto < 2 else 11 - resto)

def gera_cpf_valido():
    """Gera um número de CPF válido para fins de estudo e teste."""
    nove_digitos = ''.join(random.choice('0123456789') for _ in range(9))
    digito1 = calcula_digito(nove_digitos)
    dez_digitos = nove_digitos + digito1
    digito2 = calcula_digito(dez_digitos)
    cpf_valido = nove_digitos + digito1 + digito2
    return cpf_valido

if __name__ == "__main__":
    cpf_gerado = gera_cpf_valido()
    print(f"CPF Válido Gerado: {cpf_gerado}")

    # Opcional: Formatar o CPF para melhor visualização
    cpf_formatado = f"{cpf_gerado[:3]}.{cpf_gerado[3:6]}.{cpf_gerado[6:9]}-{cpf_gerado[9:]}"
    print(f"CPF Válido Gerado: {cpf_formatado}")

File: JavaScript/gera_cpf.js
function calculaDigito(digitos) {
    let soma = 0;
    for (let i = 0; i < digitos.length; i++) {
      soma += parseInt(digitos[i]) * (digitos.length + 1 - i);
    }
    const resto = soma % 11;
    return resto < 2 ? '0' : String(11 - resto);
  }
  
  function geraCPFValido() {
    /** Gera um número de CPF válido para fins de estudo e teste. */
    let noveDigitos = '';
    for (let i = 0; i < 9; i++) {
      noveDigitos += Math.floor(Math.random() * 10);
    }
    const digito1 = calculaDigito(noveDigitos);
    const dezDigitos = noveDigitos + digito1;
    const digito2 = calculaDigito(dezDigitos);
    const cpfValido = noveDigitos + digito1 + digito2;
    return cpfValido;
  }
  
  // Exemplo de uso:
  const cpfGerado = geraCPFValido();
  console.log(`CPF Válido Gerado: ${cpfGerado}`);
  
  // Opcional: Formatar o CPF para melhor visualização
  const cpfFormatado = `${cpfGerado.slice(0, 3)}.${cpfGerado.slice(3, 6)}.${cpfGerado.slice(6, 9)}-${cpfGerado.slice(9)}`;
  console.log(`CPF Válido Gerado: ${cpfFormatado}`);
  